flexmeasures/data/services/scheduling.py

```diff
@@ -319,28 +361,32 @@ def make_schedule(
     return True
 
 
-def find_scheduler_class(sensor: Sensor) -> type:
+def find_scheduler_class(entity: Sensor | Asset) -> type:
     """
-    Find out which scheduler to use, given a sensor.
+    Find out which scheduler to use, given a sensor or sensor.
     This will morph into a logic store utility, and schedulers should be registered for asset types there,
     instead of this fixed lookup logic.
     """
+
+    asset = entity if isinstance(entity, Asset) else entity.generic_asset
+
     # Choose which algorithm to use  TODO: unify loading this into a func store concept
-    if "custom-scheduler" in sensor.attributes:
-        scheduler_specs = sensor.attributes.get("custom-scheduler")
+    if "custom-scheduler" in entity.attributes:
```

Comment:
This would be a breaking change. Before we would first search for the `"custom-scheduler"` attribute on the sensor, whereas now we'd skip the sensor and look for the attribute on the asset straight away. Note that [our documentation](https://flexmeasures.readthedocs.io/en/latest/plugin/customisation.html#adding-your-own-scheduling-algorithm) tells users to put this attribute on the sensor.

Reply:
Good point! I looked into it and modified the code to stick to the previous behavior:

1) Look if `"custom-scheduler"` is defined as a `asset_or_sensor` attribute.

If `asset_or_sensor` is a `Sensor`, asset = `asset_or_sensor .generic_asset`.

2) Look for the asset type name